### ISSUE TITLE
Dose it support HMAC Signature to authenticate API requests?

### ISSUE THREAD

**Issue by Rwing** [ThreeMammals/Ocelot#472]
Like as aws api
https://docs.aws.amazon.com/AWSECommerceService/latest/DG/rest-signature.html

**Comment by TomPallister** [comment]
@Rwing this isn't supported at the moment.

Can I ask what use case you want this feature for?

**Comment by Rwing** [comment]
just it's simpler than jwt, oauth 🤣

**Comment by Rwing** [comment]
Is there any plan to add this feature?

**Comment by TomPallister** [comment]
@Rwing yes sure, it’s on the list!

**Comment by MikeFH** [comment]
@TomPallister Would this be materialized as an `AuthenticationHandler<T>` ?
Also would it use the Amazon way of signing requests or another method like https://github.com/w3c-dvcg/http-signatures ?

**Comment by TomPallister** [comment]
@MikeFH yes it would almost certainly be some kind of .net authentication handler. 

I haven't looked at this issue yet. I would imagine someone has already created a middleware for HMAC signing in asp.net and I would just use this. Infact I don't even think it needs to be anything to do with Ocelot......you just need to add it as the first middleware before Ocelot. I will leave this open for now because it might be nice to provide tighter integration or something.
